frontend/src/Components/Form.js

```diff
@@ -0,0 +1,161 @@
+  const dispatch = useDispatch();
+
+  const createUser = event => {
+    event.preventDefault();
+    // const USER_URL = "http://localhost:8080/users";
+    const USER_URL = "https://user-authorisation.herokuapp.com/users";
+
+    fetch(USER_URL, {
+      method: "POST",
+      body: JSON.stringify({ username, password }),
+      headers: { "Content-Type": "application/json" },
+    })
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        } //eslint-disable-next-line
+        throw "Could not create user"; // we want to display error message from backend here
+      })
+      .then(json => {
+        dispatch(
+          user.actions.setAccessToken({ accessToken: json.accessToken })
+        );
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+        localStorage.setItem("accessToken", json.accessToken);
+        setUsername("");
+        setPassword("");
+      })
+      .catch(err => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: err }));
+      });
+  };
+
+  const loginUser = event => {
+    event.preventDefault();
+    // const LOGIN_URL = "http://localhost:8080/sessions";
+    const LOGIN_URL = "https://user-authorisation.herokuapp.com/sessions";
+    fetch(LOGIN_URL, {
+      method: "POST",
+      body: JSON.stringify({ username, password }),
+      headers: { "Content-Type": "application/json" },
+    })
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        } //eslint-disable-next-line
+        throw "Unable to sign in - Please ensure username and password are correct";
+      })
+      .then(json => {
+        dispatch(
+          user.actions.setAccessToken({ accessToken: json.accessToken })
+        );
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+        localStorage.setItem("accessToken", json.accessToken);
+        setUsername("");
+        setPassword("");
+      })
+      .catch(err => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: err }));
+      });
+  };
+
+  const revealSecret = event => {
+    event.preventDefault();
+    // const SECRETS_URL = "http://localhost:8080/secrets";
+    const SECRETS_URL = "https://user-authorisation.herokuapp.com/secrets";
+    fetch(SECRETS_URL, {
+      method: "GET",
+      headers: { Authorization: accessToken },
+    })
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        } //eslint-disable-next-line
+        throw "Could not get information. Make sure you are logged in and try again.";
+      })
+      .then(json => {
+        dispatch(
+          user.actions.setSecretMessage({
+            secretMessage: JSON.stringify(json.secretMessage),
+          })
+        );
+      })
+      .catch(err => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: err }));
+      });
+  };
+
+  const signOut = () => {
+    dispatch(user.actions.logOut());
+  };
```

Comment:
Right now you are only setting userId and accessToken back to initialState. 
You should also reset secretMessage and errorMessage. Right now the this is how its working: 

1. I'm suppose to login but clicks create account instead - errorMessage appear. 
2. I login instead and reveals the secretMessage
3. I signOut 

Outcome of not resetting errorMessage : errorMessage displayed in step 1 is still there

Outcome of not resetting secretMessage: when I log in again, the secretMessage is still revealed.

Reply:
Great Tips!